src/main/java/io/github/fablabsmc/fablabs/api/fluidvolume/v1/volume/vanilla/BucketFluidVolume.java

```diff
@@ -0,0 +1,14 @@
+package io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.vanilla;
+
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.access.BucketItemAccess;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.math.Fraction;
+import io.github.fablabsmc.fablabs.api.fluidvolume.v1.volume.ImmutableFluidVolume;
+
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+
+public class BucketFluidVolume extends ImmutableFluidVolume /*unfortunately, due to the nature of items in mc and how potions are implemented, it must be immutable*/ {
```

Comment:
ahhhh, this is what the immutability is for. I think that immutability would be much preferred by the folks who most care about the shape of a fluid API right now, because immutability means easier change tracking.